src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
I'm skeptical about there being any significant performance issue here, since this code isn't on the fast path.  But regardless, yes, I'm fine with using slice_hash_table once #8842 is merged.

My concern here is really about avoiding unnecessary wheel reinvention.  As long as we don't do that, I don't really care which of the existing wheels we use. :)